docs/articles/graphql.md

```diff
@@ -0,0 +1,739 @@
+| [All images in repo](#all-images-in-repo) | ImageList | repository | image list | Returns all images in the specified repo |
+| [Global search](#global-search) | GlobalSearch | query | image summary / repo summary / layer summary | Will return what's requested in the query argument |
+| [Derived image list](#search-derived-images) | DerivedImageList | image | image list | Returns a list of images that depend on the image specified in the argument |
+| [Base image list](#search-base-images) | BaseImageList | image | image list | Returns a list of images that the specified image depends on |
+| [Get details of a specific image](#get-details-of-a-specific-image) | Image | image | image summary | Returns details about a specific image |
+| [Get referrers of a specific image](#get-referrers-of-a-specific-image) | Referrers | repo, digest, type | artifact manifests | Returns a list of artifacts of given type referring to a specific repo and digests |
+
+<a name="examples"></a>
+## Examples of zot searches using GraphQL
+
+> :pencil2: These examples show only the GraphQL query without details on how to send them to a server. See [How to use GraphQL for search queries](#howto).
+
+The query structures shown in these examples request all fields allowed by the schema for the particular query type. The schema allows you to request a subset of the data, if desired, omitting any fields that you don't need.
+
+
+### List CVEs of given image
+
+**Sample request**
+
+```graphql
+{
+  CVEListForImage(image: "centos:8", requestedPage: {limit: 1, offset:1, sortBy: SEVERITY}) {
+    Tag
+    Page {
+      TotalCount
+      ItemCount
+    }
+    CVEList {
+      Id
+      Title
+      Description
+      Severity
+      PackageList {
+        Name
+        InstalledVersion
+        FixedVersion
+      }
+    }
+  }
+}
+```
+
+**Sample response**
+
+```json
+{
+  "data": {
+    "CVEListForImage": {
+      "Tag": "8",
+      "Page": {
+        "TotalCount": 292,
+        "ItemCount": 1
+      },
+      "CVEList": [
+        {
+          "Id": "CVE-2022-24407",
+          "Title": "cyrus-sasl: failure to properly escape SQL input allows an attacker to execute arbitrary SQL commands",
+          "Description": "In Cyrus SASL 2.1.17 through 2.1.27 before 2.1.28, plugins/sql.c does not escape the password for a SQL INSERT or UPDATE statement.",
+          "Severity": "HIGH",
+          "PackageList": [
+            {
+              "Name": "cyrus-sasl-lib",
+              "InstalledVersion": "2.1.27-5.el8",
+              "FixedVersion": "2.1.27-6.el8_5"
+            }
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+### Search images affected by a given CVE id
+
+**Sample request**
+
+```graphql
+{
+  ImageListForCVE(id: "CVE-2018-20651") {
+    Results{
+      RepoName
+      Tag
+      Digest
+      ConfigDigest
```

Comment:
Some of these requests seem to be out of date. Let us check and get back to you.

Reply:
See the updated queries: https://github.com/project-zot/zot/pull/2038
There are some changes to the structures (to multiple queries in this doc, and in my review)

Reply:
Updated queries